server/models/Mention.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  DataType,
+  BelongsTo,
+  ForeignKey,
+  Column,
+  Table,
+  AllowNull,
+  AfterCreate,
+  DefaultScope,
+} from "sequelize-typescript";
+import type { SaveOptions } from "sequelize/types";
+import Document from "./Document";
+import Event from "./Event";
+import User from "./User";
+import IdModel from "./base/IdModel";
+import Fix from "./decorators/Fix";
+
+@DefaultScope(() => ({
+  include: [
+    {
+      model: User,
+      as: "user",
+    },
+    {
+      model: User,
+      as: "mentionedUser",
+    },
+    {
+      model: Document,
+      as: "document",
+    },
+  ],
+}))
+@Table({ tableName: "mentions", modelName: "mention" })
+@Fix
+class Mention extends IdModel {
```

Comment:
I wasn't expecting an entire new model, what took you down this path… perhaps it can be avoided with some discussion. I think it's probably better to keep the source of truth inside the document content.

Reply:
Hmm...TBH, I just referenced the `Backlink` implementation. But now that you raised this point, I'll revisit to understand if this could be avoided.